enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,180 @@
+- SDK tool using [operator-framework/api][oper-api] to provide the suite of tests `operator-sdk bundle validate ./bundle --select-optional name=openshiftcatalogue`
+- Pipeline using `operator-sdk bundle validate ./bundle --select-optional name=openshiftcatalogue` such as used the `operatorhub` validator via SDK currently
+
+### Non-Goals
+
+- Add checks which are not specific to `OpenShift/OKD` and that could be part of `OperatorHubValidator` to this check
+
+## Proposal
+
+### User Stories
+
+- As an operator developer, I'd like to use the SDK tool to validate my operator bundle, which will be published on OCP Catalogue so that, I could quickly get its errors and warnings locally to be able to provide the required fixes and improvements before try to publish it
+- As a member of the Pipeline team, I'd like to use the SDK tool to validate the bundles so that, I could easily avoid issues in the projects published on the catalogue
+- As an operator framework maintainer, I'd like to see the `OpenshiftCatalogueValidator` in [operator-framework/api][oper-api] so that, I can easily keep it maintained and use it in any project and tool which please me.
+- As an operator framework maintainer, I'd like to see CVP checking the operator bundle criteria when a user pushes a Pull Request to add the operator bundle on the OCP catalogue so that, we can easily ensure better quality of the projects before they are published.
+
+### Implementation Details/Notes/Constraints 
+
+Following bellow the check criteria which are proposed here for the new validator. Note that new rules and checks could be addressed in the future.
+
+#### OpenShift version supportability features
+
+The operator framework already has implemented some features that allow users to define the OCP version(s) that the operator bundle will support already. Also, another one which is the following annotation is planned to be implemented.
+
+**Annotation maxOpenShiftVersion**
+
+See [here](/enhancements/olm/max-openshift-versions-for-operators.md) a proposal for the operator bundles can use the following annotation:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    # Prevent upgrades to OpenShift Version 4.9
+    operators.coreos.com/maxOpenShiftVersion: "4.8"
+```
+
+**OCP Labels in the image index**
+
+By using the label `LABEL com.redhat.openshift.versions`, it is possible to define range criteria see:
+
+```yaml
+FROM scratch
+
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=your-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=your-channel
+LABEL operators.operatorframework.io.bundle.channel.default.v1=your-channel
+LABEL com.redhat.openshift.versions: v4.5-v4.7
```

Comment:
this could be used to decide what validation checks to perform on the bundle(which ocp versions to validate against), but this is *not* a mechanism for an operator team to constrain what versions of OCP their operator can run on.  Nothing prevents a customer from taking a 4.6 catalog and installing it on a 4.8 ocp cluster (they may or may not actually be able to install the operators that exist in that catalog, of course)

So while it's true that we should not put an operator into the 4.6 catalog if it fails validation on 4.6, and it's true that the pipeline won't put an operator into the 4.6 catalog if this version label doesn't include 4.6, those things are not the same as making an explicit statement about what versions of OCP the operator runs on.  At best it's a hint.

Reply:
Note that the checks which use this option are: 
- To verify if the operator bundle version has not any restriction and is actually configured to be installed in ANY OCP/K8S version (return a warning with the goal to only notify the users and let them beware of)
- To ensure that the range values informed for this option are valid
- To check if the operator bundle is using an API version that is deprecated or not supported in the range/versions informed when it is used and the --version suggested by you to check the config for the intend catalogue version was not provided. 

Also, I supplement the summary with the info added here by you in this comment. So, I think it is fine now.